### PR TITLE
Speed up loading DBSNP file in CollectSequencingArtifactMetrics with …

### PR DESCRIPTION
**Pull request by nh13**
…an IntervalList.

This also adds logging to the loading of DBSNP, which is useful if we
have a large set of regions or analyzing the whole genome.

@yfarjoun, @droazen, or @lbergelson care to take a look?

### CODE REVIEW

File: src/main/java/picard/analysis/artifacts/CollectSequencingArtifactMetrics.java [hunk 1]
```diff
@@ -192,10 +194,16 @@ protected void setup(final SAMFileHeader header, final File samFile) {
...
+            intervalMask = new IntervalListReferenceSequenceMask(intervalList);
+
+            if (DB_SNP != null) {
+                IOUtil.assertFileIsReadable(DB_SNP);
+                dbSnpMask = new DbSnpBitSetUtil(DB_SNP, header.getSequenceDictionary(), EnumSet.noneOf(VariantType.class), intervalList, Optional.of(log));
+            }
+        }
+        else if (DB_SNP != null) {
             IOUtil.assertFileIsReadable(DB_SNP);
             dbSnpMask = new DbSnpBitSetUtil(DB_SNP, header.getSequenceDictionary());
```
Comment (yfarjoun) [inline src/main/java/picard/analysis/artifacts/CollectSequencingArtifactMetrics.java]: why not log this case too?

**Comment by yfarjoun** [comment]
👍 looks fine. address my logging comment if you like.

File: src/main/java/picard/analysis/artifacts/CollectSequencingArtifactMetrics.java (hunk 1 above)
Comment (nh13) [inline src/main/java/picard/analysis/artifacts/CollectSequencingArtifactMetrics.java]: Good idea, will do.

**Comment by nh13** [comment]
Merging after tests pass

**Pull request merged**